bigframes/core/__init__.py

```diff
@@ -461,6 +470,16 @@ def shape(self) -> typing.Tuple[int, int]:
         width = len(self.columns)
         count_expr = self._to_ibis_expr(ordering_mode="unordered").count()
         sql = self._session.ibis_client.compile(count_expr)
+        api_methods_len = len(self._api_methods)
+
+        # Initialize methods to add as an empty list
+        add_api_methods = []
+        api_methods_len = len(self._api_methods)
+        if api_methods_len >= MAX_LABELS_COUNT:
+            add_api_methods = self.api_method[1:]
+        else:
+            add_api_methods = self.api_method
+        add_api_methods.append("shape")
```

Comment:
As Trevor said, this feels like very much the wrong layer at which to log. We should be logging public user-facing APIs not our internal methods.

Reply:
There are a lot of methods using aggregate. In this cl, I only added the methods in DataFrameGroupBy, and the basic in DataFrame/Series as example. We can follow up adding other methods in another cl.